05.ls/ls.rb

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+require_relative 'simple_formatter'
+require_relative 'detailed_formatter'
+
+def main
+  # オプション
+  option = {}
+  optparse = OptionParser.new do |opts|
+    opts.on('-a', '--all', 'output all files') { option[:a] = true }
+    opts.on('-l', '--long', 'also outputs file information') { option[:l] = true }
+    opts.on('-r', '--reverse', 'output files in reverse order') { option[:r] = true }
+  end
+
+  optparse.parse!(ARGV)
+
+  # 指定されたディレクトリを呼ぶ
+  # ディレクトリの指定がなければカレントディレクトリを呼ぶ
+  if ARGV.empty?
+    show_file_information(Dir.pwd, option)
+  else
+    directories = []
+    files = []
+    separate_directories_or_files(ARGV, directories, files)
```

Comment:
出力用の引数を渡すよりも下のように複数の戻り値を受け取るようなメソッドの方がベターです。

```suggestion
    directories, files = separate_directories_or_files(ARGV)
```

出力専用の引数を多用するとプログラムの見通しが悪くなるのでなるべく避けるようにしましょう。